logging/log_config.go

```diff
@@ -13,17 +13,46 @@ type LoggerPatternConfig struct {
 }
 
 const (
+	// pattern matching on loggers.
 	validLoggerSectionName             = `[a-zA-Z0-9]+([_-]*[a-zA-Z0-9]+)*`
 	validLoggerSectionNameWithWildcard = `(` + validLoggerSectionName + `|\*)`
 	validLoggerSections                = validLoggerSectionName + `(\.` + validLoggerSectionName + `)*`
 	validLoggerSectionsWithWildcard    = validLoggerSectionNameWithWildcard + `(\.` + validLoggerSectionNameWithWildcard + `)*`
 	validLoggerName                    = `^` + validLoggerSectionsWithWildcard + `$`
+
+	// resource configurations.
+	validNamespacePattern       = `([\w-]+|\*)`
+	validResourceTypePattern    = `(service|component|\*)`
+	validResourceSubTypePattern = validNamespacePattern
+	validModelNamePattern       = validNamespacePattern
+	validTypeSubsectionPattern  = `(` + validResourceTypePattern + `:` + validResourceSubTypePattern + `|remote:)`
+	validResourcePattern        = `^rdk.` + validNamespacePattern + `:` + validTypeSubsectionPattern + `\/` + validModelNamePattern + `$`
```

Comment:
:melting_face:

Reply:
lol

Reply:
(for the record I'm fine with this, just being dramatic)